lib/rubocop/cop/performance/hash_each_methods.rb

```diff
@@ -55,17 +67,32 @@ def autocorrect(node)
           caller, first_method = *receiver
 
           lambda do |corrector|
-            if receiver.method?(:hash)
-              correct_hash(node, corrector)
-            else
+            if first_method.eql?(:keys) || first_method.eql?(:values)
+              return correct_implicit(node, corrector) if receiver
```

Comment:
What is the predicate `receiver.child_nodes.first.nil?`?

Maybe there is a node extension method that can be used to clarify this? Or it can be extracted into a named method? 🙂